## Exercício 1

**Importando bibliotecas**

In [1]:
import numpy as np
import pandas as pd

**Importando dados como um dataframe**

In [2]:
dataframe = pd.read_fwf('data.txt',header=None,
names=["Router1", "Router2", "Router3", "Router4","Router5","Router6","Router7", "Room"])

---

### Resposta
A forma mais simples é observar qual dos roteadores tem a maior intensidade de potência média na sala correspondente.<br>
Então realizando uma média nos roteadores:

### Sala 1

In [3]:
ex1Room1 = dataframe.query('Room==1') # Pegando os dados da sala 1 
ex1Room1.mean()

Router1   -62.490
Router2   -56.276
Router3   -60.520
Router4   -64.182
Router5   -70.262
Router6   -82.920
Router7   -84.026
Room        1.000
dtype: float64

MédiaRouter1 = -64 -64 -63 ... /500 = -62.490<br>
MédiaRouter2 = -56 -57 -60 ... /500 = -56.276<br>
MédiaRouter3 = -61 -61 -60 ... /500 = -60.520<br>
MédiaRouter4 = -66 -65 -67 ... /500 = -64.182<br>
MédiaRouter5 = -71 -71 -76 ... /500 = -70.262<br>
MédiaRouter6 = -82 -85 -85 ... /500 = -82.920<br>
MédiaRouter7 = -81 -85 -84 ... /500 = -84.026

**Conclusão**:<br>
Então observando os resultados acima, podemos observar que o **roteador 2 é mais intenso na sala 1**.

---

### Sala 2

In [4]:
ex1Room2 = dataframe.query('Room==2') # Pegando os dados da sala 2 
ex1Room2.mean()

Router1   -36.924
Router2   -56.108
Router3   -55.902
Router4   -38.066
Router5   -67.614
Router6   -72.644
Router7   -73.498
Room        2.000
dtype: float64

MédiaRouter1 = -42 -44 -41 ... /500 = -36.924<br>
MédiaRouter2 = -53 -55 -58 ... /500 = -56.108<br>
MédiaRouter3 = -62 -61 -56 ... /500 = -55.902<br>
MédiaRouter4 = -38 -41 -40 ... /500 = -38.066<br>
MédiaRouter5 = -66 -66 -73 ... /500 = -67.614<br>
MédiaRouter6 = -65 -72 -69 ... /500 = -72.644<br>
MédiaRouter7 = -69 -68 -73 ... /500 = -73.498

**Conclusão**: <br>
Então observando os resultados acima, podemos concluir que o **roteador 1 é mais intenso na sala 2**.

---

### Sala 3

In [5]:
ex1Room3 = dataframe.query('Room==3') # Pegando os dados da sala 3
ex1Room3.mean()

Router1   -49.726
Router2   -54.886
Router3   -52.784
Router4   -50.702
Router5   -63.232
Router6   -81.368
Router7   -82.392
Room        3.000
dtype: float64

MédiaRouter1 = -48 -54 -50 ... /500 = -49.726<br>
MédiaRouter2 = -54 -53 -56 ... /500 = -54.886<br>
MédiaRouter3 = -50 -54 -54 ... /500 = -52.784<br>
MédiaRouter4 = -49 -50 -50 ... /500 = -50.702<br>
MédiaRouter5 = -61 -63 -71 ... /500 = -63.232<br>
MédiaRouter6 = -81 -79 -79 ... /500 = -81.368<br>
MédiaRouter7 = -84 -77 -77 ... /500 = -82.392

**Conclusão**:<br>
Então observando os resultados acima, podemos concluir que o **roteador 1 é mais intenso na sala 3**.

---

### Sala 4

In [6]:
ex1Room4 = dataframe.query('Room==4') # Pegando os dados da sala 4
ex1Room4.mean()

Router1   -60.182
Router2   -55.224
Router3   -50.650
Router4   -61.316
Router5   -49.454
Router6   -87.008
Router7   -86.990
Room        4.000
dtype: float64

MédiaRouter1 = -58 -61 -55 ... /500 = -60.182<br>
MédiaRouter2 = -56 -52 -50 ... /500 = -55.224<br>
MédiaRouter3 = -47 -49 -51 ... /500 = -50.650<br>
MédiaRouter4 = -62 -56 -61 ... /500 = -61.316<br>
MédiaRouter5 = -36 -46 -48 ... /500 = -49.454<br>
MédiaRouter6 = -85 -84 -82 ... /500 = -87.008<br>
MédiaRouter7 = -84 -83 -79 ... /500 = -86.990

**Conclusão**:<br>
Então observando os resultados acima, podemos concluir que o **roteador 5 é mais intenso na sala 4**.